src/python/pants/option/global_options.py

```diff
@@ -128,6 +128,9 @@ def register_bootstrap_options(cls, register):
     logging.addLevelName(logging.WARNING, 'WARN')
     register('-l', '--level', choices=['debug', 'info', 'warn'], default='info', recursive=True,
              help='Set the logging level.')
+    # TODO: make `-v -v -v` -> options.verbosity == 3.
+    register('-v', '--verbosity', type=int, default=0, advanced=True,
+             help='Set the verbosity level for extended debug logging.')
```

Comment:
This might have interesting interactions with the existing `-v`, `-V`, `--version` handling from the arg splitter. My guess is that `-v` will print the version and `-vv+` will change the verbosity due to how it detects the version flag.

https://github.com/pantsbuild/pants/blob/327a8d989f986623b258d5926142333d80697e4c/src/python/pants/option/arg_splitter.py#L92